chia/wallet/dao_wallet/dao_wallet.py

```diff
@@ -0,0 +1,2268 @@
+            self.wallet_state_manager, self.standard_wallet, cat_tail.hex()
+        )
+        dao_cat_wallet_id = new_dao_cat_wallet.wallet_info.id
+        dao_info = dataclasses.replace(
+            self.dao_info, cat_wallet_id=cat_wallet.id(), dao_cat_wallet_id=dao_cat_wallet_id
+        )
+        await self.save_info(dao_info)
+
+        # add treasury id to interested puzzle hashes. This is hinted in funding coins so we can track them
+        await self.wallet_state_manager.add_interested_puzzle_hashes([self.dao_info.treasury_id], [self.id()])
+        return self
+
+    @staticmethod
+    async def create(
+        wallet_state_manager: Any,
+        wallet: Wallet,
+        wallet_info: WalletInfo,
+        name: Optional[str] = None,
+    ) -> DAOWallet:
+        """
+        Create a DID wallet based on the local database
+        :param wallet_state_manager: Wallet state manager
+        :param wallet: Standard wallet
+        :param wallet_info: Serialized WalletInfo
+        :param name: Wallet name
+        :return:
+        """
+        self = DAOWallet()
+        self.log = logging.getLogger(name if name else __name__)
+        self.wallet_state_manager = wallet_state_manager
+        self.wallet_info = wallet_info
+        self.wallet_id = wallet_info.id
+        self.standard_wallet = wallet
+        self.dao_info = DAOInfo.from_json_dict(json.loads(wallet_info.data))
+        self.dao_rules = get_treasury_rules_from_puzzle(self.dao_info.current_treasury_innerpuz)
+        return self
+
+    @classmethod
+    def type(cls) -> WalletType:
+        return WalletType.DAO
+
+    def id(self) -> uint32:
+        return self.wallet_info.id
+
+    async def set_name(self, new_name: str) -> None:
+        new_info = dataclasses.replace(self.wallet_info, name=new_name)
+        self.wallet_info = new_info
+        await self.wallet_state_manager.user_store.update_wallet(self.wallet_info)
+
+    def get_name(self) -> str:
+        return self.wallet_info.name
+
+    async def match_hinted_coin(self, coin: Coin, hint: bytes32) -> bool:
+        raise NotImplementedError("Method not implemented for DAO Wallet")  # pragma: no cover
+
+    def puzzle_hash_for_pk(self, pubkey: G1Element) -> bytes32:
+        raise NotImplementedError("puzzle_hash_for_pk is not available in DAO wallets")  # pragma: no cover
+
+    async def get_new_p2_inner_hash(self) -> bytes32:
+        puzzle = await self.get_new_p2_inner_puzzle()
+        return puzzle.get_tree_hash()
+
+    async def get_new_p2_inner_puzzle(self) -> Program:
+        return await self.standard_wallet.get_new_puzzle()
+
+    def get_parent_for_coin(self, coin: Coin) -> Optional[LineageProof]:
+        parent_info = None
+        for name, ccparent in self.dao_info.parent_info:
+            if name == coin.parent_coin_info:
+                parent_info = ccparent
+        return parent_info
+
+    async def get_frozen_amount(self) -> uint64:
+        return uint64(0)  # pragma: no cover
```

Comment:
Does this function need to exist?  It's not part of the wallet protocol and seems like it's unused/pointless

Reply:
removed